### ISSUE TITLE
Encoding error in Gitlab 5.2

### ISSUE THREAD

**Issue by RubingHan** [gitlabhq/gitlabhq#4120]
I've pushed some files (named in Chinese) and when I'm trying to view the commit list, I'm getting error 500.

Here's error log:

```
Encoding::CompatibilityError (incompatible character encodings: UTF-8 and ASCII-8BIT):
  app/models/commit.rb:69:in `method_missing'
  app/contexts/commit_load_context.rb:23:in `execute'
  app/controllers/commit_controller.rb:11:in `show'

  Rendered errors/encoding.html.haml within layouts/errors (0.2ms)
  Rendered layouts/_head.html.haml (2.1ms)
  Rendered layouts/_search.html.haml (69.3ms)
  Rendered layouts/_head_panel.html.haml (74.9ms)
  Rendered layouts/_flash.html.haml (0.1ms)
Completed 500 Internal Server Error in 206ms (Views: 77.4ms | ActiveRecord: 4.4ms)
```

**Comment by xfalcox** [comment]
I'm getting this on git blame view on GitLab 7.7.0

**Comment by Razer6** [comment]
@xfalcox I think this has been fixed in GitLab 7.8 (blame view).

**Comment by DouweM** [comment]
A fix to this issue has been merged and will be in version 7.9. Please reopen this issue if the problem persists after upgrading to 7.9.

**Comment by senz** [comment]
Getting this on 8.1.2
```
Encoding::CompatibilityError (incompatible character encodings: ASCII-8BIT and UTF-8):
